### PR TITLE
Add GROMACS 2018.3

### PR DESCRIPTION
**Pull request by bkmgit**

### CODE REVIEW

File: var/spack/repos/builtin/packages/gromacs/package.py [hunk 1]
```diff
@@ -24,6 +24,7 @@ class Gromacs(CMakePackage):
     git      = 'https://github.com/gromacs/gromacs.git'
 
     version('develop', branch='master')
+    version('2018.3', 'c82634a31d0ec7dc8a128f404149440e')
```
Comment (healther) [inline var/spack/repos/builtin/packages/gromacs/package.py]: could you provide the sha256 hash?

**Comment by bkmgit** [comment]
The sum is:
4423a49224972969c52af7b1f151579cea6ab52148d8d7cbae28c183520aa291
md5sum seems to be the one listed, and is also what is on the main download page:
http://manual.gromacs.org/documentation/2018.3/download.html
http://manual.gromacs.org/documentation/2018.2/download.html

**Comment by healther** [comment]
I meant as the hash in the version statement :)

**Comment by bkmgit** [comment]
Ok. Can replace all md5sums with sha256sums. Any reason for choosing sha256 over sha512, sha384, and sha22? Is it ok to add OpenCL build as well for the later versions?

**Comment by healther** [comment]
I'm not sure what the reason for the decision for sha256 is. Maybe @tgamblin or @scheibelp have a rational for you

**Comment by adamjstewart** [comment]
Superseded by #9828.

**Pull request closed without merging**